mllib/src/main/scala/org/apache/spark/ml/tree/impl/NodeIdCache.scala

```diff
@@ -95,7 +95,7 @@ private[spark] class NodeIdCache(
       splits: Array[Array[Split]]): Unit = {
     if (prevNodeIdsForInstances != null) {
       // Unpersist the previous one if one exists.
-      prevNodeIdsForInstances.unpersist()
+      prevNodeIdsForInstances.unpersist(false)
```

Comment:
Is this change required? Should not the call to `unpersist` remain blocking?

Reply:
This is not required, but it is usually safe to unpersist without blocking in MLLIB.